tests/integration/test_route53resolver.py

```diff
@@ -0,0 +1,570 @@
+                ResolverEndpointId=create_resolver_endpoint_res["Id"],
+                TargetIps=[
+                    {"Ip": "10.0.1.200", "Port": 123},
+                ],
+            )
+
+        msg = "Resolver rules can only be associated to OUTBOUND resolver endpoints."
+        snapshot.add_transformer(snapshot.transform.key_value("Message", msg))
+        snapshot.add_transformer(snapshot.transform.jsonpath("$..Error.Message", msg))
+        snapshot.match("invalid_request_ex", inavlid_request.value.response)
+
+        # clean up
+        cleanups.append(
+            lambda: route53resolver_client.delete_resolver_endpoint(
+                ResolverEndpointId=create_resolver_endpoint_res["Id"]
+            )
+        )
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(paths=["$..SecurityGroupIds", "$..ShareStatus"])
+    def test_multipe_create_resolver_rule(
+        self, ec2_client, route53resolver_client, cleanups, snapshot
+    ):
+        request_id = short_uid()
+        resolver_endpoint_name = f"rs-{request_id}"
+        result = route53resolver_client.create_resolver_endpoint(
+            CreatorRequestId=request_id,
+            SecurityGroupIds=self.get_security_group_ids(ec2_client),
+            Direction="OUTBOUND",
+            Name=resolver_endpoint_name,
+            IpAddresses=self.get_create_resolver_endpoint_ip_address(ec2_client, "18"),
+        )
+
+        create_resolver_endpoint_res = result.get("ResolverEndpoint")
+
+        if self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        ):
+            create_resolver_endpoint_res["Status"] = "OPERATIONAL"
+
+        snapshot.match("create_resolver_endpoint_res", create_resolver_endpoint_res)
+
+        rslvr_rule_req_ids = [short_uid(), short_uid(), short_uid()]
+        for ind, req_id in enumerate(rslvr_rule_req_ids):
+            result = route53resolver_client.create_resolver_rule(
+                CreatorRequestId=req_id,
+                RuleType="FORWARD",
+                DomainName="www.example.com",
+                ResolverEndpointId=create_resolver_endpoint_res["Id"],
+                TargetIps=[
+                    {"Ip": "10.0.1.100", "Port": 123},
+                ],
+            )
+
+            create_resolver_rule_res = result.get("ResolverRule")
+            snapshot.match(f"create_resolver_rule_res_{ind}", create_resolver_rule_res)
+
+            resolver_rule_id = create_resolver_rule_res["Id"]
+            resolver_endpoint_id = create_resolver_endpoint_res["Id"]
+
+            cleanups.append(
+                lambda: route53resolver_client.delete_resolver_rule(ResolverRuleId=resolver_rule_id)
+            )
+
+        # clean up
+        cleanups.append(
+            lambda: route53resolver_client.delete_resolver_endpoint(
+                ResolverEndpointId=resolver_endpoint_id
+            )
+        )
```

Comment:
should be moved up to the line where `create_resolver_endpoint_res` is defined and placed immediately after it.